pkg/middleware/middleware.go

```diff
@@ -72,6 +74,28 @@ func Process(ctx *quaycontext.QuayRegistryContext, quay *v1.QuayRegistry, obj cl
 		return dep, nil
 	}
 
+	// If the current object is a PVC, check for volume override
+	if pvc, ok := obj.(*corev1.PersistentVolumeClaim); ok {
+		var volumeSizeOverride *resource.Quantity
+		switch quayComponentLabel {
+		case "postgres":
+			volumeSizeOverride = getVolumeSizeOverrideForComponent(quay, v1.ComponentPostgres)
+		case "clair-postgres":
+			volumeSizeOverride = getVolumeSizeOverrideForComponent(quay, v1.ComponentClair)
+		}
+
+		// If override was provided
+		if volumeSizeOverride != nil {
+			// Ensure that volume size is not being reduced
+			if pvc.Spec.Resources.Requests.Storage() != nil && volumeSizeOverride.Cmp(*pvc.Spec.Resources.Requests.Storage()) == -1 {
+				return nil, fmt.Errorf("cannot shrink volume override size from %s to %s", volumeSizeOverride.String(), pvc.Spec.Resources.Requests.Storage().String())
```

Comment:
Aren't `from` and `to` inverted here?

Reply:
@jonathankingfc this is the last one, I promise :-)

Reply:
Fixed!